### PR TITLE
Improve kubevirt conformance tests for 4.13

### PR DESCRIPTION
**Pull request by davidvossel**
This modifies the conformance tests to utilize the new base domain passthrough feature that merged into hypershift recently here, https://github.com/openshift/hypershift/pull/1787

### CODE REVIEW

**Comment by davidvossel** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-main-periodics-4.13-e2e-conformance-kubevirt

**Comment by davidvossel** [comment]
related to #35519

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-4.12-e2e-conformance-kubevirt

File: ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml [hunk 1]
```diff
@@ -41,65 +31,6 @@ chain:
...
-        port:
-          targetPort: https-443
-        to:
-          kind: Service
-          name: apps-ingress
-          weight: 100
-      EOF
-
       echo "Waiting for all VMIs in ${CLUSTER_NAMESPACE} to reach vmi.status.phase=Running"
       oc wait -n ${CLUSTER_NAMESPACE} VirtualMachineInstance --all --for=jsonpath='{.status.phase}'=Running --timeout 30m
```
Comment (LiangquanLi930) [inline ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml:35]:
```
+ echo 'Waiting for all VMIs in clusters-kv-nested to reach vmi.status.phase=Running'
Waiting for all VMIs in clusters-kv-nested to reach vmi.status.phase=Running
+ oc wait -n clusters-kv-nested VirtualMachineInstance --all '--for=jsonpath={.status.phase}=Running' --timeout 30m
error: no matching resources found
```
We should add some checks before oc wait

File: ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml [hunk 2]
```diff
@@ -41,65 +31,6 @@ chain:
       echo "Cluster became available, creating kubeconfig"
       bin/hypershift create kubeconfig --name=${KUBEVIRT_CLUSTER_NAME} >${SHARED_DIR}/nested_kubeconfig
 
-      # Enable wildcard routes on the management cluster
-      oc patch ingresscontroller -n openshift-ingress-operator default --type=json -p \
-        '[{ "op": "add", "path": "/spec/routeAdmission", "value": {wildcardPolicy: "WildcardsAllowed"}}]'
-
-      # Waiting for the default NodePort service to be created on the hosted cluster
```
Comment (LiangquanLi930) [inline ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml:48]: Do we need to keep it?

Comment (davidvossel) [reply ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml:48]: The reason to keep this logic would be if you all need to test using clusters <= 4.12.  anything >= 4.13 no longer needs this logic, but <= 4.12 does

File: ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml (hunk 1 above)
Comment (davidvossel) [reply ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml]: ah, yes i see that breaks. We probably don't even need to wait for the VMIs to come online. Waiting on the nodepool should be sufficient. I updated the code to simply print out the vmi status for debug while we wait for nodepool to finish.

**Comment by davidvossel** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-main-periodics-4.13-e2e-conformance-kubevirt

File: ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml (hunk 2 above)
Comment (davidvossel) [reply ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml:48]: yes, we do need to keep this. I didn't realize we have 4.12 periodics as well that will neeed to execute this

**Comment by davidvossel** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-main-periodics-4.13-e2e-conformance-kubevirt

**Comment by davidvossel** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-main-periodics-4.13-e2e-conformance-kubevirt

File: ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml (hunk 2 above)
Comment (LiangquanLi930) [reply ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml:48]: sorry, i have question, if https://github.com/openshift/hypershift/pull/1787 cherry-pike 4.12, we need keep it?

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-4.12-e2e-conformance-kubevirt

File: ci-operator/jobs/openshift/hypershift/openshift-hypershift-main-periodics.yaml [hunk 3]
```diff
@@ -16,13 +16,6 @@ periodics:
     ci.openshift.io/generator: prowgen
     pj-rehearse.openshift.io/can-be-rehearsed: "true"
   name: periodic-ci-openshift-hypershift-main-periodics-4.13-conformance-agent-ovn
-  reporter_config:
```
Comment (LiangquanLi930) [inline ci-operator/jobs/openshift/hypershift/openshift-hypershift-main-periodics.yaml:19]: We removed some slack notifications

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.12-periodics-4.12-e2e-conformance-kubevirt

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-main-periodics-4.13-e2e-conformance-kubevirt

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-main-periodics-4.13-e2e-conformance-kubevirt

**Pull request merged**